addons/point_of_sale/models/pos_order.py

```diff
@@ -923,6 +923,10 @@ def get_sale_details(self, date_start=False, date_stop=False, config_ids=False,
                 domain = AND([domain, [('config_id', 'in', config_ids)]])
 
         orders = self.env['pos.order'].search(domain)
+        return self._prepare_sale_details(orders, domain, date_start, date_stop, config_ids, session_ids)
```

Comment:
@jco-odoo we have used this method here, https://github.com/odoo/odoo/pull/59955/files#diff-ddb00c6e8c73e1deb14b52b9eadcce65ab644d10a857216c0eb5d0adb7d70f3dR12
because we need some extra details for the report.

Reply:
@bat-odoo and why could not you just override the original method?

Reply:
Hello @jco-odoo 
Actually we need config_ids and order details for further computation, otherwise we need to browse orders again.
That's why this method is distributed in two parts to take out order details.

Thanks

Reply:
@pimodoo As we need the first and last pos order, it is not something provided here.  What is better: the refactoring or adding posorders in the return dict?